### PR TITLE
feat(clean-css): Make clean-css a peer depedency

### LINKED ISSUE #6: clean css as a peer dependency?
Hi! Since this metalsmith plugin wraps clean-css, maybe we could put clean-css as a peer dependency to allow users to upgrade it without you having to upgrade your own lib? I am looking at using your plugin but wanted to ask that.

I can do the PR :)

### PR DESCRIPTION
**Pull request by vvo**
fixes #6 

... And a lot of other things completely unrelated.

Hi @aymericbeaumet, this PR contains:
- move to yarn workspaces. It allows requiring metalsmith-clean-css in
example easily and other quick wins like simpler travis setup
- add documentation on how to develop and release (quick interactive release script)
- upgrade all packages
- prepare renovate configuration, it will update all packages minor and
patch without you having to notic, to activate it: https://github.com/algolia/renovate-config-algolia#3-activate-renovate
- remove node4 testing
- simplify travis configuration, there should no more be needed that
many configs
- make clean-css a peer dependency
- change code to make it work with clean-css 4.1.9

Going further, to make this repo "modern hipster friendly", I would
switch testing framework to Jest and do a snapshot on the resulting
build folder of Metalsmith example.

Let me know what you think.

### CODE REVIEW

**Comment by vvo** [comment]
Not sure why you would want appveyor builds, since there's no specific code for windows or linux

**Comment by vvo** [comment]
@aymericbeaumet PING, any comment? If you think that's really not the way you want to move, no problem I can only change some parts

File: .appveyor.yml [hunk 1]
```diff
@@ -1,35 +1,21 @@
 environment:
   matrix:
```
Comment (aymericbeaumet) [inline .appveyor.yml]: Open question: shouldn't the node.js maintenance LTS (v4) be included in the matrix until [april 2018](https://github.com/nodejs/Release)?

File: .travis.yml [hunk 2]
```diff
@@ -1,32 +1,16 @@
...
-
-script:
-  - node --version && npm --version
-  - npm test
-  - pushd example && node build.js && popd
+  - 8
+  - 9
 
 after_success:
   - npm run codeclimate
```
Comment (aymericbeaumet) [inline .travis.yml]: `yarn codeclimate`?

File: package.json [hunk 3]
```diff
@@ -1,31 +1,32 @@
...
-  "name": "metalsmith-clean-css",
-  "version": "5.0.1",
-  "description": "A Metalsmith plugin to minify CSS files",
-  "keywords": [
-    "metalsmith",
-    "plugin"
-  ],
+  "name": "metalsmith-clean-css-monorepo",
+  "private": true,
   "author": "Aymeric Beaumet <aymeric@beaumet.me>",
```
Comment (aymericbeaumet) [inline package.json]: Feel free to add yourself as collaborator 👍

File: package.json [hunk 4]
```diff
@@ -1,31 +1,32 @@
...
+  "extends": [
```
Comment (aymericbeaumet) [inline package.json]: I don't know about this?

File: package.json [hunk 5]
```diff
@@ -1,31 +1,32 @@
...
+    "ava": "0.24.0",
+    "codeclimate-test-reporter": "0.5.0",
+    "nyc": "11.3.0",
+    "standard": "10.0.3"
   },
   "scripts": {
-    "test": "standard && nyc --reporter=html --check-coverage --branches 100 --functions 100 --lines 100 --statements 100 ava",
-    "codeclimate": "nyc report --reporter=text-lcov | codeclimate-test-reporter"
+    "build-example": "node packages/metalsmith-clean-css-example/build.js",
+    "dev": "ava packages/metalsmith-clean-css/test --watch",
```
Comment (aymericbeaumet) [inline package.json]: I usually find the `yarn test:watch` semantic clearer, dev could mean a lot of things (like starting a project with livereload)

File: .appveyor.yml (hunk 1 above)
Comment (aymericbeaumet) [reply .appveyor.yml]: (this should not be an issue with [Buffer.from](https://nodejs.org/docs/latest-v4.x/api/buffer.html#buffer_class_method_buffer_from_str_encoding))

File: package.json (hunk 4 above)
Comment (vvo) [reply package.json]: https://renovateapp.com/docs/ you'll have to activate it afterwards. It will auto update dependencies in your project.

**Comment by vvo** [comment]
Yo @aymericbeaumet I just pushed some changes given your comments.

Only thing I will not do and don't think we need to do is the jest move. I have too little experience dealing with jest and codeclimate/codecoverage so that would take me a lot of time.

It's good to go now I believe so I can't trash that post-it of you from my physical desk at Algolia :D

See ya :)

**Comment by vvo** [comment]
(Do add me to npm publishing and github rights so I can test that release scripts)

File: .travis.yml (hunk 2 above)
Comment (vvo) [reply .travis.yml]: updated

File: .appveyor.yml (hunk 1 above)
Comment (vvo) [reply .appveyor.yml]: As you want but I prefer not supporting those very old Node.js versions and encourage people to upgrade (Node 4 = IE6 of server side)

File: package.json (hunk 5 above)
Comment (vvo) [reply package.json]: updated

File: package.json (hunk 3 above)
Comment (vvo) [reply package.json]: I am good :D

File: .appveyor.yml (hunk 1 above)
Comment (aymericbeaumet) [reply .appveyor.yml]: Sounds good 👌

File: package.json (hunk 4 above)
Comment (aymericbeaumet) [reply package.json]: Nice!

**Comment by aymericbeaumet** [comment]
Thanks for the awesome work 🎉 Just gave you the rights you need on GitHub/npm. Feel free to merge, bump, publish, etc...

You will have more free space on your screen hopefully :D

See you soon 👋

**Comment by aymericbeaumet** [comment]
🎉

**Pull request merged**